lib/castle/api.rb

```diff
@@ -2,44 +2,32 @@
 
 module Castle
   # this class is responsible for making requests to api
-  class API
-    def initialize(headers = {})
-      @config = Castle.config
-      @http = prepare_http
-      @headers = headers.merge('Content-Type' => 'application/json')
-    end
-
-    def request(command)
-      request = Castle::Request.new(@headers).build(
-        command.path,
-        command.data,
-        command.method
-      )
-      perform_request(request)
-    end
-
-    private
+  module API
+    EXCEPTIONS = [
+      Timeout::Error,
+      Errno::EINVAL,
+      Errno::ECONNRESET,
+      EOFError,
+      Net::HTTPBadResponse,
+      Net::HTTPHeaderSyntaxError,
+      Net::ProtocolError
+    ].freeze
 
-    def prepare_http
-      http = Net::HTTP.new(@config.host, @config.port)
-      http.read_timeout = @config.request_timeout / 1000.0
-      prepare_http_for_ssl(http) if @config.port == 443
-      http
-    end
-
-    def prepare_http_for_ssl(http)
-      http.use_ssl = true
-      http.verify_mode = OpenSSL::SSL::VERIFY_PEER
-    end
+    class << self
+      def request(command, headers = {})
+        raise Castle::ConfigurationError, 'configuration is not valid' unless Castle.config.valid?
 
-    def perform_request(request)
-      raise Castle::ConfigurationError, 'configuration is not valid' unless @config.valid?
-      begin
-        Castle::Response.new(@http.request(request)).parse
-      rescue Timeout::Error, Errno::EINVAL, Errno::ECONNRESET, EOFError,
-             Net::HTTPBadResponse, Net::HTTPHeaderSyntaxError,
-             Net::ProtocolError
-        raise Castle::RequestError, 'Castle API connection error'
+        begin
+          Castle::API::Response.call(
+            Castle::API::Request.call(
+              command,
+              Castle.config.api_secret,
+              headers
+            )
+          )
+        rescue *EXCEPTIONS
```

Comment:
we should wrap the original error in our handler the same way we do it within the tcp core so we can use it if needed